### PR TITLE
Worker: skip over functions, check strict mode

### PR DESCRIPTION
**Pull request by marcoscaceres**
The current test assumes that everything in Navigator is going to be an attribute. 

The badging API adds some (IDL) operations, which are writable. 

Oh, also, tests for strict mode were missing.

### CODE REVIEW

File: workers/interfaces/WorkerUtils/navigator/007.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 var log = [];
 var neverEncounteredValue = "This is not the value you are looking for.";
 for (x in navigator) {
+  // skip functions, as they are settable
```
Comment (annevk) [inline workers/interfaces/WorkerUtils/navigator/007.js:4]: I'm not sure what settable means here in ES terms. Writable/configurable?

File: workers/interfaces/WorkerUtils/navigator/008.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<div id=log></div>
+<script>
+async_test(function() {
+  var worker = new Worker('008.js');
+  worker.onmessage = this.step_func(function(e) {
+    assert_equals(e.data, '');
+    this.done();
+  });
+});
+</script>
```
Comment (annevk) [inline workers/interfaces/WorkerUtils/navigator/008.html:14]: Newline at the end of the file, but please also use https://web-platform-tests.org/writing-tests/testharness.html#dedicated-worker-tests-worker-js instead. That'll be much clearer.

File: workers/interfaces/WorkerUtils/navigator/007.js (hunk 1 above)
Comment (marcoscaceres) [inline workers/interfaces/WorkerUtils/navigator/007.js:4]:
```suggestion
  // skip functions, as they are writable
```

File: workers/interfaces/WorkerUtils/navigator/008.worker.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  let log = [];
+  for (const x in navigator) {
+    // skip functions, as they are settable
+    if (typeof navigator[x] === "function") continue;
+
+    // Trying to override a property should throw in strict mode
+    try {
+      navigator[x] = "";
+      log.push(x);
+    } catch (err) {}
```
Comment (annevk) [inline workers/interfaces/WorkerUtils/navigator/008.worker.js:13]: This needs to become an assert_throws_js or some such, no?

Comment (marcoscaceres) [reply workers/interfaces/WorkerUtils/navigator/008.worker.js:13]: yeah, that's better.

**Review by annevk — APPROVED** [review]

**Pull request merged**